File: 0917-reverse-only-letters/0917-reverse-only-letters.java
class Solution {
    private boolean isLetter(char c) {
        return Character.isLetter(c);
    }
    public String reverseOnlyLetters(String s) {
        char[] arr = s.toCharArray();
        int l = 0, r = arr.length - 1;

        while (l < r) {
            if (isLetter(arr[l]) && isLetter(arr[r])) {
                char temp = arr[l];
                arr[l] = arr[r];
                arr[r] = temp;
                l++;
                r--;
            } else if (!isLetter(arr[l])) {
                l++;
            } else { 
                r--;
            }
        }
        return new String(arr);
    }
}

File: 0917-reverse-only-letters/0917-reverse-only-letters.cpp
class Solution {
public:
    string reverseOnlyLetters(string s) {
        int n = s.size();
        int l=0,r=n-1;
        while(l < r){
            if(((s[l] >= 'a' && s[l] <= 'z') || (s[l] >= 'A' && s[l] <= 'Z')) && ((s[r] >= 'a' && s[r] <= 'z') || (s[r] >= 'A' && s[r] <= 'Z'))){
                char ch = s[l];
                s[l] = s[r];
                s[r] = ch;
                l++;
                r--;
            }
            else if(!((s[l] >= 'a' && s[l] <= 'z') || (s[l] >= 'A' && s[l] <= 'Z'))){
                l++;
            }else if(!((s[r] >= 'a' && s[r] <= 'z') || (s[r] >= 'A' && s[r] <= 'Z'))){
                r--;
            }
        }
        return s;
    }
};